src/status_im/ui2/screens/chat/group_details/view.cljs

```diff
@@ -51,38 +50,35 @@
-       (fn []
-         (let [{window-height :height} (rn/use-window-dimensions)
-               safe-area (safe-area/use-safe-area)]
-           [rn/view {:style {:height (- window-height (:top safe-area))}}
-            [rn/touchable-opacity
-             {:on-press #(rf/dispatch [:bottom-sheet/hide])
-              :style    (style/close-icon)}
-             [quo2/icon :i/close {:color (colors/theme-colors colors/neutral-100 colors/white)}]]
-            [quo2/text {:size   :heading-1
-                        :weight :semi-bold
-                        :style  {:margin-left 20}}
-             (i18n/label :t/add-members)]
-            [rn/text-input {:placeholder (str (i18n/label :t/search) "...")
-                            :style       {:height             32
-                                          :padding-horizontal 20
-                                          :margin-vertical    12}
-                            :on-change   (fn [e] (rf/dispatch [:contacts/set-search-query (oget e "nativeEvent.text")]))}]
-            [contact-list/contact-list {:icon    :check
-                                        :group   group
-                                        :added   added
-                                        :search? true}]
-            [rn/view {:style style/bottom-container}
-             [quo2/button {:style    {:flex 1}
-                           :on-press #(rf/dispatch [:bottom-sheet/hide])
-                           :disabled (zero? (count @added))}
-              (i18n/label :t/save)]]]))])))
+(defn add-members-sheet [group admin?]
+  [:f>
+   (fn []
+     (let [{window-height :height} (rn/use-window-dimensions)
+           safe-area             (safe-area/use-safe-area)
+           selected-participants (rf/sub [:selected-participants])
+           deselected-members    (rf/sub [:deselected-members])]
+       [rn/view {:style {:height (- window-height (:top safe-area))}}
+        [rn/touchable-opacity
+         {:on-press #(rf/dispatch [:bottom-sheet/hide])
+          :accessibility-label :close-manage-members
+          :style    (style/close-icon)}
+         [quo2/icon :i/close {:color (colors/theme-colors colors/neutral-100 colors/white)}]]
+        [quo2/text {:size   :heading-1
+                    :weight :semi-bold
+                    :style  {:margin-left 20}}
+         (i18n/label (if admin? :t/manage-members :t/add-members))]
+        [contact-list/contact-list {:icon    :check
+                                    :group   group
+                                    :search? true}]
+        [rn/view {:style style/bottom-container}
+         [quo2/button {:style    {:flex 1}
+                       :accessibility-label :save
+                       :on-press (fn []
+                                   (rf/dispatch [:group-chats.ui/add-members-pressed])
+                                   (rf/dispatch [:group-chats.ui/remove-members-pressed])
+                                   (rf/dispatch [:bottom-sheet/hide]))
```

Comment:
Please can you elaborate, why add and remove events simultaneously?

Reply:
An admin can select new contacts to add to group, and deselect current members to remove from group. Save button will do both updates.